lapack/testlapack/dlauu2.go

```diff
@@ -87,24 +88,24 @@ func dlauuTest(t *testing.T, dlauu func(blas.Uplo, int, []float64, int), uplo bl
 
 			// Compute U*Uᵀ or Lᵀ*L using Dgemm with U and L
 			// represented as dense triangular matrices.
-			ldwant := n
-			want := make([]float64, n*ldwant)
+			r := cloneGeneral(blas64.General{Rows: n, Cols: n, Data: a, Stride: lda})
 			if uplo == blas.Upper {
 				// Use aCopy as a dense representation of the upper triangular U.
 				u := aCopy
 				ldu := lda
-				// Compute U * Uᵀ and store the result into want.
+				// Compute U*Uᵀ - A and store the result into R.
 				bi.Dgemm(blas.NoTrans, blas.Trans, n, n, n,
-					1, u, ldu, u, ldu, 0, want, ldwant)
+					1, u, ldu, u, ldu, -1, r.Data, r.Stride)
```

Comment:
Nice.

Reply:
Fortrans's TESTING directory is a useful source of learning material.